libcloud/common/base.py

```diff
@@ -46,11 +48,48 @@
 
 from libcloud.utils.misc import lowercase_keys
 from libcloud.utils.compression import decompress_data
+from libcloud.common.exception import RateLimit, exception_from_message
 from libcloud.common.types import LibcloudError, MalformedResponseError
-
 from libcloud.httplib_ssl import LibcloudHTTPConnection
 from libcloud.httplib_ssl import LibcloudHTTPSConnection
 
+DEFAULT_TIMEOUT = 3
```

Comment:
This is really low for a default timeout, especially given that it's retrying.  I'd think more like 30 would be sensible as a default.  It has to account for both the sleep between retries and the time it spent waiting for the API to respond.